group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/AbstractUniformAssignmentBuilder.java

```diff
@@ -65,4 +65,33 @@ protected static Set<TopicIdPartition> topicIdPartitions(
                 .mapToObj(i -> new TopicIdPartition(topic, i))
             ).collect(Collectors.toSet());
     }
+
+    /**
+     * Constructs a set of {@code TopicIdPartition} including all the partitions that are
+     * currently not assigned to any member.
+     *
+     * @param topicIds                      Collection of topic Ids.
+     * @param subscribedTopicDescriber      Describer to fetch partition counts for topics.
+     * @param groupSpec                The group's assignment spec.
+     *
+     *
+     * @return Set of unassigned {@code TopicIdPartition} including newly added topic partitions.
+     */
+    protected static Set<TopicIdPartition> unassignedTopicIdPartitions(
```

Comment:
I just noticed that we already have a loop in `OptimizedUniformAssignmentBuilder#buildAssignment` iterating over the subscribed topic ids. I would be great if we could merge this one into it.